deadlines/src/test/java/com/palantir/deadlines/DeadlinesTest.java

```diff
@@ -43,15 +44,22 @@ public void test_duration_to_header_value() {
         assertThat(headerValue).isEqualTo("1.523");
     }
 
-    @Test
-    public void test_header_value_to_duration() {
-        assertThat(Deadlines.tryParseSecondsToNanoseconds("1.523"))
-                .isNotNull()
-                .isEqualTo(Duration.ofMillis(1523).toNanos());
+    @ParameterizedTest
+    @CsvSource({
+        "1, 1000000000",
+        "' 2. ', 2000000000",
+        "3.0, 3000000000",
+        "3.1, 3100000000",
+        "1234567890.12345, 1234567890123450112",
+        "1.523, 1523000000",
+        "'   1.523  ', 1523000000",
+    })
+    public void test_header_value_to_duration(String input, long expectedNanos) {
+        assertThat(Deadlines.tryParseSecondsToNanoseconds(input)).isNotNull().isEqualTo(expectedNanos);
     }
 
     @ParameterizedTest
-    @ValueSource(strings = {"", " ", ",", ".", "d"})
+    @ValueSource(strings = {"", " ", ",", ".", "d", " 1,234", "-1.234", "1.234e5"})
```

Comment:
I'm not sure we should allow leading and trailing whitespace, but I don't have a strong preference

Reply:
Any leading & trailing whitespace is trimmed by JDK `Double.parseDouble` per spec.

* https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/Double.html#parseDouble(java.lang.String)
* https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/Double.html#valueOf(java.lang.String)
* https://github.com/openjdk/jdk/blob/jdk-25%2B8/src/java.base/share/classes/jdk/internal/math/FloatingDecimal.java#L1830-L1839
* https://github.com/openjdk/jdk/blob/jdk-25%2B8/src/java.base/share/classes/jdk/internal/math/FloatingDecimal.java#L109-L110
* https://github.com/openjdk/jdk/blob/jdk-25%2B8/src/java.base/share/classes/java/lang/Double.java#L971C16-L971C31

Reply:
👍